Proposals/NNNN-calendar-recurrence-rule.md

```diff
@@ -0,0 +1,466 @@
+        /// the day, but in different time zones due to a daylight saving switch
+        public enum RepeatedTimePolicy : Sendable, Codable {
+            /// Consider only the eariler date
+            case onlyFirst
+            /// Consider only the later date
+            case onlyLast
+            /// Consider both dates
+            case both
+        }
+        /// What to do when there are multiple recurrences occurring at the same
+        /// time of the day but in different time zones due to a daylight saving
+        /// transition.
+        ///
+        /// For example, an event with daily recurrence rule that starts at 1 am
+        /// on November 2 in PST will repeat on:
+        ///
+        /// - 2.11.2024 01:00 PDT (08:00 UTC)
+        /// - 3.11.2024 01:00 PDT (08:00 UTC)
+        /// - 3.11.2024 01:00 PST (09:00 UTC)
+        /// - 4.11.2024 01:00 PST (09:00 UTC)
+        ///
+        /// Due to the time zone switch on November 3, there are different times
+        /// when the event might repeat.
+        ///
+        /// Default value is `.onlyFirst`
+        public var repeatedTimePolicy: RepeatedTimePolicy
+        /// How often a recurring event repeats
+        public enum Frequency: Sendable, Codable {
+            case minutely, hourly, daily, weekly, monthly, yearly
+        }
+        /// How often the event repeats
+        public var frequency: Frequency
+        /// At what interval to repeat
+        ///
+        /// Default value is `1`
+        public var interval: Int
+        /// When a recurring event stops recurring
+        public struct End: Sendable, Codable {
+            /// The event stops repeating after a given number of times
+            /// - Parameter count: how many times to repeat the event, including
+            ///                    the first occurrence. `count` must be greater
+            ///                    than `0`
+            public static func afterOcurrences(_ count: Int) -> Self
+            /// The event stops repeating after a given date
+            /// - Parameter date: the date on which the event may last occur. No
+            ///                   further ocurrences will be found after that
+            public static func afterDate(_ date: Date) -> Self
+            /// The event repeats indefinitely
+            public static var never: Self
+        }
+        /// For how long the event repeats
+        public var end: End
+        
+        public enum Weekday: Sendable, Codable {
+            /// Repeat on every weekday
+            case every(Locale.Weekday)
+            /// Repeat on the n-th instance of the specified weekday in a month,
+            /// if the recurrence has a monthly frequency. If the recurrence has
+            /// a yearly frequency, repeat on the n-th week of the year.
+            /// 
+            /// If n is negative, repeat on the n-to-last of the given weekday.
+            case nth(Int, Locale.Weekday)
```

Comment:
I would suggest adding conveniences like:

```swift
extension Weekday {
    public static func first(_ weekday: Locale.Weekday) -> Self { .nth(1, weekday) }
    public static func second(_ weekday: Locale.Weekday) -> Self { .nth(2, weekday) }

    public static func last(_ weekday: Locale.Weekday) -> Self { .nth(-1, weekday) }
    public static func secondToLast(_ weekday: Locale.Weekday) -> Self { .nth(-2, weekday) }
}
```

These are much easier to understand that `.nth(-1, .wednesday)`.

Reply:
We can keep iterating it during the review period